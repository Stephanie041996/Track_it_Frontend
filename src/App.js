import React from 'react';
import './App.css';
import {
  Switch, Route, Redirect,
} from 'react-router-dom';
import Measurements from './containers/measurements';

function App() {
  return (
    <div className="App">
      <Switch>
        <Route path="/" exact component={Measurements} />
        
        <Redirect to="/" />
      </Switch>
    </div>
  );
}

export default App;
