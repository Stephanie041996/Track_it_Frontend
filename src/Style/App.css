.measurements {
  height: 82%;
  justify-content: center;
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: repeat(3, 8rem);
  padding-bottom: 200px;
  padding-left: 10px;
  padding-right: 10px;
}

.measurement-item {
  display: flex;
  justify-content: center;
  text-decoration: none;
  align-items: center;
  padding: 1rem;
  margin: 0.5rem;
  background-color: white;
  box-shadow: 2px 2px 5px gray;
}

.measurement-item:hover {
  box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
}

.measurement-item img {
  width: 4.5rem;
  filter:
    invert(68%) sepia(34%) saturate(764%) hue-rotate(169deg)
    brightness(94%) contrast(91%);
}

.progress-item {
  display: flex;
  justify-content: space-between;
  padding: 1rem;
  color: #696969;
  background-color: white;
  border-bottom:  1px solid #f3f4f6;
}

.progress-container {
  height: 82%;
  padding-top: 100px;
  padding-bottom: 20px;
  background-color: #f3f4f6;
}

.top-nav {
  list-style-type: none;
  position: fixed;
  top: 0;
  width: 100%;
  display: flex;
  justify-content: space-around;
  background-color: #42b5e8;
  color: white;
  padding-top: 20px;
  padding-bottom: 20px;
  z-index: 1;
}

.top-nav a {
  color: white;
  text-decoration: none;
}

.bottom-nav {
  list-style-type: none;
  position: fixed;
  bottom: 0;
  width: 100%;
  display: flex;
  justify-content: space-around;
  background-color: #464a4a;
  color: rgb(230, 230, 230);
  padding-top: 20px;
  padding-bottom: 20px;
}

.bottom-nav a {
  color: white;
  text-decoration: none;
}

.add-measure {
  height: 52%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.container {
  width: 90%;
  background: white;
  padding: 2rem 0;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.container,
button {
  width: 150px;
}

.Add-heading {
  text-align: center;
  padding-top: 100px;
}

.banner-section {
  padding-top: 70px;
  text-align: center;
}

.Login-form {
  align-items: center;
  justify-content: center;
}

.select-box {
  height: 45px;
  width: 309px;
}

.input-box {
  height: 40px;
  width: 300px;
}

.input-form {
  display: flex;
  flex-direction: column;
  align-items: center;
}

button {
  height: 45px;
  width: 309px;
  background-color: #42b5e8;
}

.home-section {
  text-align: center;
  padding-top: 150px;
}

.banner-chart img {
  width: 100%;
}

h3 {
  color: #464a4a !important;
  text-align: center;
}

.button-wrapper {
  text-align: center;
}
