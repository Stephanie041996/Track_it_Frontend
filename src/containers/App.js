import React, { Component } from 'react';
import {
  BrowserRouter, Switch, Route, Redirect,
} from 'react-router-dom';
import axios from 'axios';
import Measurements from './measurements';
import Progress from './Progress';
import AddTransaction from '../components/AddTransaction';
import Home from '../components/Home';
import Home2 from '../components/Home2';

export default class App extends Component {
  constructor() {
    super();

    this.state = {
      loggedInStatus: 'NOT_LOGGED_IN',
      user: {},
    };

    this.handleLogin = this.handleLogin.bind(this);
    this.handleLogout = this.handleLogout.bind(this);
  }

  componentDidMount() {
    this.checkLoginStatus();
  }

  handleLogout() {
    this.setState({
      loggedInStatus: 'NOT_LOGGED_IN',
      user: {},
    });
  }

  handleLogin(data) {
    this.setState({
      loggedInStatus: 'LOGGED_IN',
      user: data.user,
    });
  }

  checkLoginStatus() {
    axios
      .get('https://guarded-brushlands-05784.herokuapp.com/logged_in', { withCredentials: true })
      .then((response) => {
        if (
          response.data.logged_in
            && this.state.loggedInStatus === 'NOT_LOGGED_IN'
        ) {
          this.setState({
            loggedInStatus: 'LOGGED_IN',
            user: response.data.user,
          });
        } else if (
          !response.data.logged_in
            & (this.state.loggedInStatus === 'LOGGED_IN')
        ) {
          this.setState({
            loggedInStatus: 'NOT_LOGGED_IN',
            user: {},
          });
        }
      })
      .catch((error) => {
        console.log('check login error', error);
      });
  }

  render() {
    return (
      <div className="App">
        <BrowserRouter>
          <Switch>

            <Route exact path="/add" component={AddTransaction} />
            <Route path="/progress/:measurementId" component={Progress} />

            <Route
              exact
              path="/"
              render={(props) => (
                <Home
                  {...props}
                  handleLogin={this.handleLogin}
                  handleLogout={this.handleLogout}
                  loggedInStatus={this.state.loggedInStatus}
                />
              )}
            />
            <Route
              exact
              path="/reg"
              render={(props) => (
                <Home2
                  {...props}
                  handleLogin={this.handleLogin}
                  handleLogout={this.handleLogout}
                  loggedInStatus={this.state.loggedInStatus}
                />
              )}
            />
            <Route
              exact
              path="/dashboard"
              render={(props) => (
                <Measurements
                  {...props}
                  loggedInStatus={this.state.loggedInStatus}
                />
              )}
            />
            <Redirect to="/" />
          </Switch>
        </BrowserRouter>
      </div>
    );
  }
}
