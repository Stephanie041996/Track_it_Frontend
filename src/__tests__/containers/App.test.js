import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../../store/index';
import App from '../../containers/App';

afterEach(cleanup);

it("Takes website's snapshot", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <App />
    </Provider>,
  );
  expect(
    asFragment(
      <Provider store={store}>
        <App />
      </Provider>,
    ),
  ).toMatchSnapshot();
});

it("Displays app's title", () => {
  render(
    <Provider store={store}>
      <App />
    </Provider>,
  );
  const element = screen.getByText(/Track it/i);
  expect(element).toBeInTheDocument();
});

it('Displays username field', () => {
  render(
    <Provider store={store}>
      <App />
    </Provider>,
  );
  const element = screen.getByPlaceholderText(/username/i);
  expect(element).toBeInTheDocument();
});

it('Displays login button', () => {
  render(
    <Provider store={store}>
      <App />
    </Provider>,
  );
  const element = screen.getByText(/Login/i);
  expect(element).toBeInTheDocument();
});
