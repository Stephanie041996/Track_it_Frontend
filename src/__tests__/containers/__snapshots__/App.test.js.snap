// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Takes website's snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="home-section"
    >
      <nav>
        <div
          class="top-nav"
        >
          <a
            href="/add"
          >
            <p>
              Track It
            </p>
          </a>
        </div>
      </nav>
      <div
        class="Login-form"
      >
        <form
          class="input-form"
        >
          <input
            class="input-box"
            name="username"
            placeholder="username"
            required=""
            type="text"
            value=""
          />
          <button
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
      <a
        href="/reg"
      >
        Not Registerd ? Register
      </a>
    </div>
  </div>
</DocumentFragment>
`;
